src/kubernetes_cluster/spec/kubernetes_api/state_machine.rs

```diff
@@ -34,6 +34,8 @@ verus! {
 //
 // + Support more operations like List
 
+impl <K: ResourceView, E: ExternalAPI, R: Reconciler<K, E>> Cluster<K, E, R> {
+
 // TODO: maybe make it a method of DynamicObjectView?
 pub open spec fn object_has_well_formed_spec(obj: DynamicObjectView) -> bool {
```

Comment:
Can we add `&&& obj.kind == CustomResource::kind() ==> K::unmarshal_spec(obj.spec).is_Ok()` here?

Reply:
That's not the problem.

Reply:
What I need is the kind of object conforms to the one of object in `etcd`. Adding such a constraint only tells that the new object can be successfully `unmarshall`ed.

Reply:
But the old one can also be successfully unmarshalled because the old request was also checked whether it was well formed. And we have the invariant saying that each object in etcd can be unmarshalled (that invariant doesn't include cr objects but it can be included). So in the proof it is easy to show that old object can also be unmarshalled.

Reply:
What does this have to do with proof? We need to pass the result of unmarshal in the state machine to `transition_rule`.

Reply:
In the spec, nothing prevents you from calling `K::unmarshall(old_obj.spec).get_Ok_0()` and you can pretend that unmarshall always returns an Ok in the spec code.
The only problem is whether we can show that unmarshall indeed returns OK for old_object to the verifier, and that invariant mentioned above solves this problem.

Reply:
It's not pretend.... I guess what you want to do is just to return an error when the unmarshal process of old_obj fails.

Reply:
Sorry I don't get you. I am just proposing to write
```
transition_rule(K::from_dynamic_object(old_obj).get_Ok_0(), K::from_dynamic_object(new_obj).get_Ok_0())
```
What would be your concerns here?

Reply:
Sorry, maybe I misunderstand something. But should we move `marshal` and `unmarshal` to `ResourceView`?